### ISSUE TITLE
BUG: `default_sort` throws an `Array to string conversion` error when defined as an `array`

### ISSUE THREAD

**Issue by pjayme** [symbiote/silverstripe-gridfieldextensions#357]
When a `DataObject` defines the `default_sort` property using the following convention:

```
private static array $default_sort = [
    'SortOrder' => 'ASC',
];
```

Then this throws an `Array to string conversion` error as [updates](https://github.com/symbiote/silverstripe-gridfieldextensions/blob/3/src/GridFieldOrderableRows.php#L401) from Feb 22’ now expects this property to be of type `string` rather than of type `array`. 

The simple solution is to use the `string` convention instead:

```
private static string $default_sort = 'SortOrder ASC';
```

I’m raising this as I’m not sure if this should be treated as a regression since the module was capable of handling the `array` convention or if the expectation is that users should simply stick to using the new `string` convention instead? Or should the module be backwards compatible in that it can handle both conventions?

## Acceptance criteria
- Default sort define as array should still work when using Gridfiled extensions.
- The bug https://github.com/symbiote/silverstripe-gridfieldextensions/pull/325 was trying to fix is not reintroduce

## Original PRs
- https://github.com/symbiote/silverstripe-gridfieldextensions/pull/352
- https://github.com/symbiote/silverstripe-gridfieldextensions/pull/349

## PRs
- https://github.com/symbiote/silverstripe-gridfieldextensions/pull/361

**Comment by sunnysideup** [comment]
This is going to break a lot of things, so I would love to see this fixed ASAP. 

It is tricky, because you have to make sure that the "local" sort is the same as the default_sort.  Probably the easiest way is to check the default_sort. If it is an array then add "local" sort (the ones compiled in the code) as a array as well and vice versa. 
https://github.com/symbiote/silverstripe-gridfieldextensions/blob/3/src/GridFieldOrderableRows.php#L379-L406

**Comment by maxime-rainville** [comment]
I think this came from https://github.com/symbiote/silverstripe-gridfieldextensions/pull/325

**Comment by maxime-rainville** [comment]
Looks like it's been causing some degree of problem:
- https://github.com/symbiote/silverstripe-gridfieldextensions/issues/344
- https://github.com/gorriecoe/silverstripe-menu/pull/17
- https://github.com/gorriecoe/silverstripe-menu/issues/18

**Comment by GuySartorelli** [comment]
Fixed in [3.5.1](https://github.com/symbiote/silverstripe-gridfieldextensions/releases/tag/3.5.1) and [3.6.1](https://github.com/symbiote/silverstripe-gridfieldextensions/releases/tag/3.6.1)
